fvm/environment/programs.go

```diff
@@ -82,7 +87,19 @@ func (programs *Programs) set(
 		return err
 	}
 
-	programs.derivedTxnData.SetProgram(address, program, state)
+	location, dependencies, err := programs.dependencyStack.pop()
+	if err != nil {
+		return err
+	}
+	if location != address {
```

Comment:
this check and the pop stack empty check is probably not necessary.  i don't feel strongly about this / feel free to keep it around.

Reply:
Added comment.